engine/base_client/client.py

```diff
@@ -110,12 +110,25 @@ def run_experiment(
                         continue
 
                 search_params = {**searcher.search_params}
+                ef = "default"
+                if "search_params" in search_params:
+                    ef = search_params["search_params"].get("ef", "default")
```

Comment:
```suggestion
                ef = search_params.get("search_params", {}).get("ef", "default")
```

Although, `search_params` key inside `serach_params` var is ugly. Better to change that in the experiment configs soon.